torch/testing/_internal/opinfo/core.py

```diff
@@ -262,6 +262,24 @@ def to_noncontiguous(t):
         return self.transform(to_noncontiguous)
 
 
+def make_sample_input(
+    input: Union[torch.Tensor, Sequence[torch.Tensor]], *args, **kwargs
```

Comment:
Input isn't always a tensor or sequence of tensors anymore, which puts the code in an awkward place as its primacy doesn't make so much sense anymore. For now I think we could just model this as `Any`. 

I think adding an ability like this is very natural and I was thinking along similar lines. Do you think instead of adding this function we can change the constructor to `SampleInput` so developers can just write:

```
SampleInput(a, b, out=out)
```

just like calling the function? I understand that the SampleInput constructor accepts kwargs controlling its behavior, but I don't think these kwargs overlap with operator parameters?

```
output_process_fn_grad=lambda x: x,
broadcasts_input=False,
name="",
```

I would also like to kill the `broadcasts_input` and `name` params, but that's a future discussion. Basically I'm curious if we can update the signature for the SampleInput constructor to be:

```
(arg, *args, 
output_process_fn_grad=lambda x: x,
broadcasts_input=False,
name="",
**kwargs)
```

?

Reply:
I'm not a big fan of this because the keyword arguments meant for the operator and those meant as metadata aren't clearly separated when reading the code. The assumption of no name collisions is also something that might come back to bite use later.

How about this syntax though:
```
SampleInput(a, b, out=out).output_process_fn_grad(fn)
```

The arguments are clearly separated from the metadata and 99% of the time you just call `SampleInput` with nothing extra.

Reply:
I've updated the PR to allow `SampleInput` to be constructed with var-args and var-kwargs in a backwards-compatible way. It occurred to me that any edge cases that don't work can just fall back to the existing syntax, so it doesn't matter too much.